internal/provider/utils/filtering.go

```diff
@@ -85,25 +86,12 @@ func filterServer(server models.Server,
 	return false
 }
 
-func filterByPossibilities(value string, possibilities []string) (filtered bool) {
+func filterByPossibilities[K string | uint16](value K, possibilities []K) (filtered bool) {
```

Comment:
I think (don't get mad at me 🙏) it was better before. Using `strings.EqualFold(fmt.Sprint(value), fmt.Sprint(possibility))` is kinda confusing. What we could do however is change `filterByPossibilitiesUint16` to take in a `T comparable` and use `==` instead of `strings.EqualFold` so it could extend to other comparable types. We should keep a separate string focused function which ignores casing.